templates/account/partials/login_form.html

```diff
@@ -19,27 +19,33 @@
       <a class="link--styled" href="{% url 'account_reset_password' %}">
         {% trans "Forgot password?" context "Login form secondary link" %}
       </a>
-      {% with available_backends=settings.available_backends %}
-      <hr>
-      <div class="row">
-        <div class="col-sm-6 col-md-12 col-xl-6">
-          {% if 'facebook' in backends.backends and 'facebook' in available_backends %}
-            <a class="btn social-btn facebook" href="{% url 'social:begin' 'facebook' %}">
-              <svg data-src="{% static "images/facebook_logo.svg" %}" width="18" height="18" />
-              <small>{% trans 'Facebook' %}</small>
-            </a>
-          {% endif %}
-        </div>
-        <div class="col-sm-6 col-md-12 col-xl-6">
-          {% if 'google-oauth2' in backends.backends and 'google-oauth2' in available_backends %}
-            <a class="btn social-btn google" href="{% url 'social:begin' 'google-oauth2' %}">
-              <svg data-src="{% static "images/google_logo.svg" %}" width="18" height="18" />
-              <small>{% trans 'Google' %}</small>
-            </a>
-          {% endif %}
-        </div>
-      </div>
-    {% endwith %}
+      {% if settings.available_backends %}
+        {% with available_backends=settings.available_backends %}
```

Comment:
If we changed the order of the above two lines to:
```
{% with available_backends=settings.available_backends %}
    {% if available_backends %}
```
the second statement would use cached `available_backends` (that's what the `{% with %}` tag does).

Reply:
Didn't know, cool